content/automate/ManageIQ/Infrastructure/VM/Transform/Import.class/__methods__/list_storages.rb

```diff
@@ -10,25 +10,29 @@ def initialize(handle = $evm)
               end
 
               def main
+                multi_vm = @handle.root['vm'].nil?
+
                 values_hash = {}
                 values_hash[nil] = '-- select storage from list --'
 
-                provider_id = @handle.root['dialog_provider']
-                @handle.log(:info, "Selected provider: #{provider_id}")
-                if provider_id.present? && provider_id != '!'
-                  provider = @handle.vmdb(:ext_management_system, provider_id)
-                  if provider.nil?
-                    values_hash[nil] = 'None'
-                  else
-                    provider.storages.each do |storage|
-                      values_hash[storage.id] = storage.name
+                unless multi_vm
+                  provider_id = @handle.root['dialog_provider']
+                  if provider_id.present? && provider_id != '!'
```

Comment:
My curiosity, what does it mean if `provider_id == '!'`?

Reply:
@bzwei Actually I have no idea. I didn't write this code.